--- a/configure.ac
+++ b/configure.ac
@@ -116,10 +116,8 @@ AS_IF([test "x$enable_tests" != "xno"],
 		  json-glib-1.0 >= $JSON_GLIB_REQUIRED_VERSION
 		  gio-2.0 >= $GIO_REQUIRED_VERSION,
 		  [have_tests=yes]
-)
-PKG_CHECK_MODULES(DBUSMENUTESTSVALGRIND, valgrind, have_valgrind=yes, have_valgrind=no)
-AM_CONDITIONAL([HAVE_VALGRIND], [test "x$have_valgrind" = "xyes"])
-])
+)], [have_tests=no])
+AM_CONDITIONAL([HAVE_VALGRIND], [false])
 
 AC_SUBST(DBUSMENUTESTS_CFLAGS)
 AC_SUBST(DBUSMENUTESTS_LIBS) 